test_util/provide_aws.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+import logging
+import os
+
+import boto3
+import retrying
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+# Allows adjusting the timeout in CI
+AWS_WAITING_TIMEOUT = os.getenv('AWS_WAITING_TIMEOUT', 60)
+
+
+class BotoWrapper():
+    def __init__(self, region, aws_access_key_id, aws_secret_access_key):
+        self.region = region
+        self.session = boto3.session.Session(
+                aws_access_key_id=aws_access_key_id, aws_secret_access_key=aws_secret_access_key)
+
+    def client(self, name):
+        return self.session.client(service_name=name, region_name=self.region)
+
+    def resource(self, name):
+        return self.session.resource(service_name=name, region_name=self.region)
+
+    def create_key_pair(self, key_name):
+        """Retruns private key of newly generated pair
+        """
+        key = self.client('ec2').create_key_pair(KeyName=key_name)
+        return key['KeyMaterial']
+
+    def delete_key_pair(self, key_name):
+        self.client('ec2').delete_key_pair(KeyName=key_name)
+
+    def create_stack(self, name, template_url, user_parameters):
+        """Returns boto stack object
+        """
+        logging.info('Requesting AWS CloudFormation...')
+        cf_parameters = []
+        for k, v in user_parameters.items():
+            cf_parameters.append({'ParameterKey': k, 'ParameterValue': v})
+        return self.resource('cloudformation').create_stack(
+                StackName=name,
+                TemplateURL=template_url,
+                DisableRollback=True,
+                TimeoutInMinutes=30,
+                Capabilities=['CAPABILITY_IAM'],
+                Parameters=cf_parameters)
+
+
+class CFStack():
+    def __init__(self, region, aws_access_key_id, aws_secret_access_key):
```

Comment:
This should take two parts:
1. the stack name
2. a boto wrapper

Making this have all the parameters of botowrapper means that as we build / change botowrapper we have to change this vs. just "construct a botowrapper any way you can"

The constructor taking a stack name or stack id means thatthe CFStack object always can have a `self.stack`, and the constructor both works for the "new one" case as well as the "attach" case, both of which put the object into the same state.